Accepts `PathLike` objects for dataset readers

damienpontifex opened the pull request:
Continuation from #17465
* Dataset operations convert `PathLike` objects to string
* `compat.path_to_str` now accepts one or more objects to convert
* Unit tests

Further work to #15784 to allow `PathLike` objects to be passed to Dataset readers as well.

damienpontifex committed:
Accepts `PathLike` objects for dataset readers

* Dataset operations convert `PathLike` objects to string
* `compat.path_to_str` now accepts one or more objects to convert

damienpontifex committed:
work on testing compat functions

damienpontifex committed:
Fix for tensor circular import and dtype import in test

olicht commented on tensorflow/python/util/compat_test.py:
```diff
@@ -0,0 +1,78 @@
+        transformed = compat.path_to_str(tensor_input)
+        self.assertEqual(tensor_input, transformed)
+
+    
```
Please add newline at the end of the file

damienpontifex replied:
Thanks, removed whitespace so it's newline at end of file

damienpontifex committed:
end of file as newline

damienpontifex commented:
@qlzh727 how are the required checks triggered here? They are listed as required but still waiting, is this usual

qlzh727 commented:
Lint is failing for this change. See more details in https://source.cloud.google.com/results/invocations/00af55ab-8a63-48e9-9ddf-22e9764663d4/log. 

Mostly its related to py formet, eg, 4 space indent vs 2 space indent.

damienpontifex committed:
pylint fixes

damienpontifex commented:
Thanks @qlzh727 
Ran pylint on python file changes and fixed issues. There were still three notices on tensorflow/python/util/compat.py of:
```
tensorflow/python/util/compat.py:58:2: R1705: Unnecessary "elif" after "return" (no-else-return)
tensorflow/python/util/compat.py:80:2: R1705: Unnecessary "elif" after "return" (no-else-return)
tensorflow/python/util/compat.py:109:2: R1705: Unnecessary "else" after "return" (no-else-return)
```
I hadn't made changes to these lines in this PR so didn't change. 

All my changes for compat.py, compat_test.py and readers.py are now fixed for linting

mrry commented on tensorflow/python/util/compat_test.py:
```diff
@@ -0,0 +1,77 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Nit: s/2017/2018/

damienpontifex replied:
😄 Copy and paste job. Updated to 2018

mrry commented on tensorflow/python/util/compat_test.py:
```diff
@@ -0,0 +1,77 @@
+
+  def test_transforms_pathlib_path(self):
+    if Path is None:
+      return # Skip test if pathlib not available.
```
Use `self.skipTest("Test requires Python 3.6 or greater.")` here and below.

damienpontifex replied:
Thanks, I've replaced return statements in both tests targeting pathlib

mrry reviewed (changes requested):
Just a couple of minor nits, but otherwise this looks good, thanks!

damienpontifex committed:
It's 2018 and proper way to skip tetsts

mrry reviewed (dismissed):
Looks great, thanks!

damienpontifex commented:
@qlzh727 noticed a couple of checks failing:
From sanity checks (I'm not sure what the fix for these are, can you suggest?)
- 10. do_check_load_py_test: Check load py_test: Check that BUILD files with py_test target properly load py_test
- do_buildifier: buildifier check

And the XLA check? Any suggestions or changes needed before merge?

qlzh727 commented on tensorflow/python/BUILD:
```diff
@@ -3525,6 +3525,23 @@ py_test(
+        ":framework_for_generated_wrappers",
+        ":math_ops",
+        ":util",
+        "//third_party/py/numpy"
```
I don't think the np is needed here since it is not used in test.

qlzh727 commented on tensorflow/python/util/compat_test.py:
```diff
@@ -0,0 +1,77 @@
+import inspect
+
+from tensorflow.python.platform import test
+from tensorflow.python.platform import tf_logging as logging
```
The logging lib is not used in test.

qlzh727 commented on tensorflow/python/util/compat_test.py:
```diff
@@ -0,0 +1,77 @@
+  Path = None
+
+try:
+  from collections.abc import Iterable as _Iterable
```
Didn't see the _Iterable gets used anywhere here.

qlzh727 commented on tensorflow/python/BUILD:
```diff
@@ -3525,6 +3525,23 @@ py_test(
+    deps = [
+        ":array_ops",
+        ":client_testlib",
+        ":framework",
```
Please clean up the deps since some of them are not used in the test.

qlzh727 reviewed (changes requested):
Its not clear to me that why the tests are failing, but some of item might need updates.

damienpontifex committed:
cleanup for unused imports

damienpontifex commented:
Thanks @qlzh727 updated with your suggested cleanups

tensorflowbutler commented:
Nagging Assignee @qlzh727: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

damienpontifex commented:
@qlzh727 can the checks be run again to get an updated status on where this PR is at?

qlzh727 commented:
Sorry for the delay, kicking off new tests now.

damienpontifex commented:
@qlzh727 seems like some of the builds failed, but seem unrelated to the files changed in this PR. Do you have suggestions on a fix?

qlzh727 commented:
The error message was buried in the log and bit hard to find, but here it is.

=== Sanity check step 4 of 12: do_buildifier (buildifier check) ===
Running do_buildifier on 386 files
tensorflow/python/BUILD # reformat listsort unsafesort sort:py_test.deps
buildifier took 0 s
FAIL: buildifier found errors and/or warnings in above BUILD files.
buildifier suggested the following changes:
3713d3712
<         ":client_testlib",
3715,3716c3714,3716
<         ":util"
<     ]
---
>         ":client_testlib",
>         ":util",
>     ],
Please fix manually or run buildifier <file> to auto-fix.

qlzh727 commented:
it requires "," for each of the deps.

tensorflowbutler commented:
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

damienpontifex committed:
Fix buildifier errors with comma after each line

tensorflowbutler commented:
It has been 29 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

qlzh727 commented:
The PR need to rebase again. Please do so if this is still active. Thanks and sorry for the late reply.

damienpontifex committed:
fixes for tf_py_test

damienpontifex commented:
@qlzh727 thanks. I've rebased with master now. Thanks for the ongoing work on this and appreciate the feedback and involvement

tensorflowbutler commented:
It has been 28 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
